File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/channels/InvalidVibrationPatternException.java
package expo.modules.notifications.notifications.channels;

import expo.modules.core.errors.CodedRuntimeException;

public class InvalidVibrationPatternException extends CodedRuntimeException {
  public InvalidVibrationPatternException(int invalidValueKey, Object invalidValue) {
    super("Invalid value in vibration pattern, expected all elements to be numbers, got: " + invalidValue + " under " + invalidValueKey);
  }

  @Override
  public String getCode() {
    return "ERR_INVALID_VIBRATION_PATTERN";
  }
}

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/channels/InvalidVibrationPatternException.kt
package expo.modules.notifications.notifications.channels

import expo.modules.core.errors.CodedRuntimeException

class InvalidVibrationPatternException(invalidValueKey: Int, invalidValue: Any?) :
  CodedRuntimeException("Invalid value in vibration pattern, expected all elements to be numbers, got: $invalidValue under $invalidValueKey") {

  override fun getCode() = "ERR_INVALID_VIBRATION_PATTERN"
}
